mesonbuild/mesonmain.py

```diff
@@ -54,6 +54,16 @@ def setup_vsenv():
         return
     if 'Visual Studio' in os.environ['PATH']:
         return
+    # VSINSTALL is set when running setvars from a Visual Studio installation
+    # Tested with Visual Studio 2012 and 2017
+    if os.environ.get('VSINSTALLDIR', bat_placeholder) != '%VSINSTALLDIR%':
```

Comment:
Shouldn't this be `if 'VSINSTALLDIR' in os.environ`?

Reply:
While at it, I noticed that bat_placeholder is useless, `os.environ.get('VSINSTALLDIR')` with no default returns `None` which is fine to compare with a string.

Reply:
It was not leftover. [It was intentional}(https://youtu.be/kK4H-LkrQjQ?t=127).

Reply:
changed the check to "in os.environ"

Reply:
I wonder why the shutil.which checks at the beginning does not include all the entries in "default_c" from environment.py but only 'cc', 'clang' and 'clang-cl'

Reply:
> It was not leftover. [It was intentional](https://youtu.be/kK4H-LkrQjQ?t=127).

ahah, ok :)